### PR TITLE
Add RouterDevices for MQTT Connections

### PR DESCRIPTION
**Pull request by Frank-Wiebeler**
+ADD: Class to represent RouterDevices JSON Structure
+ADD: Create Mqtt Connection based on RouterDevice

### CODE REVIEW

File: agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/common/MqttOptionService.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.dke.data.agrirouter.api.env.Environment;
 import com.dke.data.agrirouter.impl.EnvironmentalService;
 import com.dke.data.agrirouter.impl.common.ssl.KeyStoreCreationService;
+import de.dev4agriculture.dto.RouterDevice;
```
Comment (saschadoemer) [inline agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/common/MqttOptionService.java]: Internal package declaration?

File: agrirouter-api-java-api/src/test/java/com/dke/data/agrirouter/api/factories/impl/ReadRouterDeviceTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package com.dke.data.agrirouter.api.factories.impl;
+
+import com.google.gson.Gson;
+import de.dev4agriculture.dto.RouterDevice;
+import org.junit.jupiter.api.Test;
+
+public class ReadRouterDeviceTest {
+
+  @Test
+  void givenValidRouterDeviceCanBeCreated() {
```
Comment (saschadoemer) [inline agrirouter-api-java-api/src/test/java/com/dke/data/agrirouter/api/factories/impl/ReadRouterDeviceTest.java]: A test without assertion is not a test. Add an meaningful assertion or remove the test.

File: agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/dto/onboard/RouterDevice.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package de.dev4agriculture.dto
+
+class RouterDevice{
+    class Authentication{
+        lateinit var type: String
+        lateinit var secret: String
+        lateinit var certificate: String
+
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/dto/onboard/RouterDevice.kt]: Added blank lines instead 😈  (now formating is in line with all the other files like e.g. OnboardingResponse.kt ;) )

File: agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/common/MqttClientService.java [hunk 4]
```diff
@@ -33,4 +34,28 @@ public IMqttClient create(OnboardingResponse onboardingResponse) {
...
+          this.environment.getMqttServerUrl(
+              routerDevice.getConnectionCriteria().getHost(),
+              String.valueOf(routerDevice.getConnectionCriteria().getPort())),
+          routerDevice.getConnectionCriteria().getClientId());
+    } catch (MqttException e) {
+      throw new CouldNotCreateMqttClientException("Could not create MQTT client.", e);
+    }
+  }
+
+  public void addEndpoint(IMqttClient mqttClient, OnboardingResponse onboardingResponse)
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/common/MqttClientService.java]: As you wish 🧞

File: agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/dto/onboard/RouterDevice.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package de.dev4agriculture.dto
```
[the remark this answers is not in the document]
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/main/kotlin/com/dke/data/agrirouter/api/dto/onboard/RouterDevice.kt]: Uups, extraction error 🙈 

Fixed, thanks for mentioning

File: agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/common/MqttOptionService.java (hunk 1 above)
Comment (Frank-Wiebeler) [reply agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/common/MqttOptionService.java]: Uups, extraction error 🙈

Fixed, thanks for mentioning

File: agrirouter-api-java-api/src/test/java/com/dke/data/agrirouter/api/factories/impl/ReadRouterDeviceTest.java (hunk 2 above)
Comment (Frank-Wiebeler) [reply agrirouter-api-java-api/src/test/java/com/dke/data/agrirouter/api/factories/impl/ReadRouterDeviceTest.java]: Removed

**Pull request merged**